packages/puppeteer-core/src/bidi/core/Request.ts

```diff
@@ -34,6 +36,8 @@ export class Request extends EventEmitter<{
     return request;
   }
 
+  #responseCompleteDeferred = Deferred.create<void, Error>();
```

Comment:
do we need a deferred here? let's just let the code to query without regard to the lifecycle. Do any tests depend on this deferred?

Reply:
Also, even if we need the waiter in the client side we should add it to the HTTPResponse and not to the core/, using rxjs to properly handle timeouts/closures/etc

Reply:
removed